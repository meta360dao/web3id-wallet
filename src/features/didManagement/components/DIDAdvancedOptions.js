import React, {useMemo} from 'react';
import {translate} from '../../../locales';
import {FormControl, Input, Select, Stack} from 'native-base';
import {InputPopover, SelectToggler} from '../../../design-system';

export function DIDAdvancedOptions({onChange, form}) {
  const keypairType = useMemo(() => {
    if (form.didType === 'diddock') {
      return [
        {
          label: translate('did_advenced_options.keypair_type_ed25519'),
          value: 'ed25519',
        },
        {
          label: translate('did_advenced_options.keypair_type_sr25519'),
          value: 'sr25519',
        },
        {
          label: translate('did_advenced_options.keypair_type_ecdsa'),
          value: 'ecdsa',
        },
      ];
    }
    return [
      {
        label: translate('did_advenced_options.keypair_type_ed25519'),
        value: 'ed25519',
      },
    ];
  }, [form.didType]);

  return (
    <SelectToggler
      placeholder={translate('account_advanced_options.advanced_options')}>
      <FormControl>
        <Stack mt={7}>
          <FormControl.Label>
            {translate('account_advanced_options.keypair_type')}
            <InputPopover
              title={translate('account_advanced_options.keypair_type')}>
              {translate('account_advanced_options.keypair_type_tip')}
            </InputPopover>
          </FormControl.Label>
          <Select
            onValueChange={onChange('keypairType')}
            selectedValue={form.keypairType}>
            {keypairType.map(({label, value}, index) => {
              return (
                <Select.Item
                  key={`keypairType_${index}`}
                  label={label}
                  value={value}
                />
              );
            })}
          </Select>
        </Stack>
      </FormControl>
      <FormControl isInvalid={false}>
        <Stack my={7}>
          <FormControl.Label>
            {translate('account_advanced_options.secret_derivation_path')}
            <InputPopover
              title={translate(
                'account_advanced_options.secret_derivation_path',
              )}>
              {translate('account_advanced_options.secret_derivation_path_tip')}
            </InputPopover>
          </FormControl.Label>
          <Input
            placeholder="//hard/soft//password"
            onChangeText={onChange('derivationPath')}
            autoCapitalize="none"
          />
          <FormControl.ErrorMessage>
            {translate('account_advanced_options.invalid_secret_uri')}
          </FormControl.ErrorMessage>
        </Stack>
      </FormControl>
    </SelectToggler>
  );
}
