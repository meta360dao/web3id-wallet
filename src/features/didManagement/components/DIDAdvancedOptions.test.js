import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {DIDAdvancedOptions} from './DIDAdvancedOptions';

const mockStore = configureMockStore();

describe('DIDAdvancedOptions', () => {
  it('should render correctly for diddock', () => {
    const initialState = {
      // placeholder for redux store
    };

    const form = {
      didType: 'diddock',
    };
    const wrapper = shallow(
      <DIDAdvancedOptions onChange={jest.fn()} form={form} />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render correctly for didkey', () => {
    const initialState = {
      // placeholder for redux store
    };

    const form = {
      didType: 'didkey',
    };
    const wrapper = shallow(
      <DIDAdvancedOptions onChange={jest.fn()} form={form} />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
