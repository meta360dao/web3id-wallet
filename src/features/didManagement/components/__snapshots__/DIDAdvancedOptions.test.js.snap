// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DIDAdvancedOptions should render correctly for diddock 1`] = `
<Memo(ForwardRef(Box))>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <Memo(ForwardRef(Stack))
      alignItems="center"
      backgroundColor="#27272A"
      borderRadius={6}
      direction="row"
      mt={8}
      pl={5}
      pr={2}
      py="1px"
      width={0.67}
    >
      <Memo(ForwardRef(Box))
        flex={1}
      >
        <Styled(Component)
          variant="label"
        >
          Advanced options
        </Styled(Component)>
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        mt={2}
      >
        <Memo(ForwardRef(createdIcon)) />
      </Memo(ForwardRef(Box))>
    </Memo(ForwardRef(Stack))>
  </TouchableWithoutFeedback>
</Memo(ForwardRef(Box))>
`;

exports[`DIDAdvancedOptions should render correctly for didkey 1`] = `
<Memo(ForwardRef(Box))>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <Memo(ForwardRef(Stack))
      alignItems="center"
      backgroundColor="#27272A"
      borderRadius={6}
      direction="row"
      mt={8}
      pl={5}
      pr={2}
      py="1px"
      width={0.67}
    >
      <Memo(ForwardRef(Box))
        flex={1}
      >
        <Styled(Component)
          variant="label"
        >
          Advanced options
        </Styled(Component)>
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        mt={2}
      >
        <Memo(ForwardRef(createdIcon)) />
      </Memo(ForwardRef(Box))>
    </Memo(ForwardRef(Stack))>
  </TouchableWithoutFeedback>
</Memo(ForwardRef(Box))>
`;
